libraries/Microsoft.Bot.Builder/SkypeMentionNormalizeMiddleware.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    ///  Middleware to patch mention Entities from Skype since they don't conform to expected values.
+    ///  Bots that interact with Skype should use this middleware if mentions are used.
+    /// </summary>
+    /// <description>
+    ///  A Skype mention "text" field is of the format:
+    ///    &lt;at id=\"28:2bc5b54d-5d48-4ff1-bd25-03dcbb5ce918\">botname&lt;/at&gt;
+    ///  But Activity.Text doesn't contain those tags and RemoveMentionText can't remove
+    ///  the entity from Activity.Text.
+    ///  This will remove the &lt;at&gt; nodes, leaving just the name.
+    /// </description>
+    public class SkypeMentionNormalizeMiddleware : IMiddleware
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SkypeMentionNormalizeMiddleware"/> class.
+        /// </summary>
+        public SkypeMentionNormalizeMiddleware()
+        {
+        }
+
+        public static void NormalizeSkypMentionText(Activity activity)
+        {
+            if (activity.ChannelId == "skype" && activity.Type == ActivityTypes.Message)
```

Comment:
This would be better as Bot.Connector.Channels.Skype instead of "skype"

Reply:
I'll defer to whatever you all think.  I wasn't sure where to put it, so included it in the same place as some other middleware.  But it had occurred to me a Channel specific package might be better.

Reply:
I'm just referring to the constant "skype" ... all channel constants are here Bot.Connector.Channels